sdk/src/main/scala/com/horizen/account/state/AccountState.scala

```diff
@@ -122,13 +123,19 @@ class AccountState(val params: NetworkParams,
 
       // get also list of receipts, useful for computing the receiptRoot hash
       val receiptList = new ListBuffer[EthereumReceipt]()
-      val blockNumber = stateView.getHeight + 1
+      val blockNumber = stateMetadataStorage.getHeight + 1
```

Comment:
Is this the correct way getting the new block number?

Reply:
I think so - @i-Alex?